### DISCUSSION TITLE
Choice of Framework for Precipitate Growth

### DISCUSSION THREAD

**Discussion by nwinner** [idaholab/moose#28792] [Q&A Modules: Phase field]
Hello all, I am new to the MOOSE community, having found it as a platform for solving phase field problems. I have some known multicomponent, multiphase alloys with CALPHAD database files. When the alloy is cast, there is a distribution of intermetallic precipitates that are of interest to me. I want to study the impact of processing temperature on the growth/coallescence of these phases. Ideally, starting with annealing, and then including different temperature profiles as well as gradients in the alloy from a cold/hot side. The precipitates are highly anisotropic, so treating the interfaces properly is important to me.

I'd like to know which phase field formalism is the best starting point for this. From the [website](https://mooseframework.inl.gov/modules/phase_field/index.html), it seems like there are a couple of different frameworks that could potentially solve this. The [sublattice KKS model](https://mooseframework.inl.gov/modules/phase_field/MultiPhase/SLKKS.html), the [multiphase model](https://mooseframework.inl.gov/modules/phase_field/MultiPhase/WBM.html), and the [grand potential model](https://mooseframework.inl.gov/modules/phase_field/MultiPhase/GrandPotentialMultiphase.html). 

The paper on SLKKS specifically discusses importing calphad models, so this seems most ideal, but I cannot see much discussion on it. There is a single example on [CrFe](https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/slkks/CrFe.i), which currently has CrFe_sigma_out_var_0001.csv file missing, so the file cannot be tested. The grand potential formalism is described as offering some improved computational performance as compared to the SL-KKS model. Since I'm working with .tdb files, SL-KKS would be more straightforward it seems, but perhaps approximating the phases with a series parabolic free energies could be worth improved performance. 

Given my description above, which would be the best framework to start with in terms of performance and ease-of-use. I will stork my own MOOSE app once I get a feel for things, but obviously I'd like a gentle introduction if possible.

In either case, are their examples out there which seem analogous to this problem?

**Comment by GiudGiud** [comment]
@laagesen 
@dschwen for this missing file :S (see previous post)

**Reply by laagesen** [reply]
I would start with Sublattice KKS and use that if you can get the performance you need. But I would use the simplest chemistry you can to start out with, try starting with a binary alloy Calphad description to work out as many kinks as possible before moving to ternary+ alloys.

**Reply by nwinner** [reply]
Got it. I'll plan to start with SLKKS and Ni-Cu or something. Are there other working examples for SLKKS while we wait for this missing file?

**Reply by nwinner** [reply]
@GiudGiud and @dschwen it actually looks like the missing file is provided by first running CrFe_sigma.i, which produces the csv files as output. See this short [PR](https://github.com/idaholab/moose/pull/28802#issue-2571698722).

**Reply by GiudGiud** [reply]
awesome I ll take a look at the end of the week
